include/sensors/core/tasks/capacitive_sensor_task.hpp

```diff
@@ -0,0 +1,154 @@
+        writer.write(configuration_data, ADDRESS);
+        configuration_data = FDC_CONFIGURATION << 8 | SAMPLE_RATE;
+        writer.write(configuration_data, ADDRESS);
+    }
+
+  private:
+    void visit(std::monostate &m) {}
+
+    void visit(can_messages::ReadFromSensorRequest &m) {
+        LOG("Received request to read from %d sensor\n", m.sensor);
+        if (bool(m.offset_reading)) {
+            capdac_callback.reset();
+            capdac_callback.set_offset(CAPDAC_OFFSET);
+            writer.write(CONFIGURATION_MEASUREMENT, ADDRESS);
+            sensor_callbacks::SingleRegisterCallback callback{capdac_callback};
+            writer.read(ADDRESS, callback, callback, CONFIGURATION_MEASUREMENT);
+        } else {
+            capacitance_callback.reset();
+            capacitance_callback.set_offset(CAPDAC_OFFSET);
+            sensor_callbacks::MultiRegisterCallback callback{capacitance_callback};
+            writer.multi_register_poll(ADDRESS, 1, DELAY, callback, callback,
+                                       MSB_MEASUREMENT_1, LSB_MEASUREMENT_1);
+        }
+    }
+
+    void visit(can_messages::WriteToSensorRequest &m) {
+        LOG("Received request to write data %d to %d sensor\n", m.data,
+            m.sensor);
+        writer.write(m.data, ADDRESS);
+    }
+
+    void visit(can_messages::BaselineSensorRequest &m) {
+        LOG("Received request to read from %d sensor\n", m.sensor);
+        if (bool(m.offset_update)) {
+            capdac_callback.reset();
+            capdac_callback.set_number_of_reads(m.sample_rate);
+            capdac_callback.set_offset(CAPDAC_OFFSET);
+            sensor_callbacks::SingleRegisterCallback callback{capdac_callback};
+            writer.single_register_poll(ADDRESS, m.sample_rate, DELAY,
+                                        callback, callback,
+                                        CONFIGURATION_MEASUREMENT);
+        } else {
+            capacitance_callback.reset();
+            capacitance_callback.set_number_of_reads(m.sample_rate);
+            capacitance_callback.set_offset(CAPDAC_OFFSET);
+            sensor_callbacks::MultiRegisterCallback callback{capacitance_callback};
+            writer.multi_register_poll(ADDRESS, m.sample_rate, DELAY,
+                                       callback, callback, MSB_MEASUREMENT_1,
+                                       LSB_MEASUREMENT_1);
+        }
+    }
+
+    void visit(can_messages::SetSensorThresholdRequest &m) {
+        LOG("Received request to set threshold to %d from %d sensor\n",
+            m.threshold, m.sensor);
+        THRESHOLD = m.threshold;
+        auto message = can_messages::SensorThresholdResponse{
+            {}, SensorType::capacitive, THRESHOLD};
+        can_client.send_can_message(can_ids::NodeId::host, message);
+    }
+
+    InternalCallback internal_callback{};
+    sensor_task_utils::BitMode mode = sensor_task_utils::BitMode::MSB;
+    // 8 pF
+    uint32_t THRESHOLD = 0x8;
```

Comment:
These three are in all caps. Are they constants?

Reply:
Sort of..you can change the threshold to something else from the python side but it shouldn't be changed frequently.

Reply:
Ok. Can we make them lower case or `const`?